### PR TITLE
feat: Client UI translate to Filament (from React)

### PR DESCRIPTION
**Pull request by lancepioch**
Edit By Charles
Below are things that need to be done...Good Luck!

## Profile 
 - [x]  2FA
   	 - [x] Enable
   	 - [x] Disable
   	 - [x] Backup codes
  - [x] Timezone Changing
  - [x] 12/24hr Time Changing
  - [x] Change Password
  - [x] Change Email
  - [x] Change Language
  - [x]  API
	 - [x] Create API Keys
	 - [x] Delete API Keys
	 - [x] Last Used Timestamp
 - [x] SSH Keys
	 - [x] Adding of SSH Keys
	 - [x] Deleting of SSH Keys
 - [x] Activity Logs
 - [x] OAuth Link/ Unlink

## Server List

Needs to be redone... I'll just leave it at that...

## Server Area

 - [x] Console Page
	 - [x] Console
	 - [x] Power Actions
	 - [x] Stats + Graphs
 - [x] File Manager
	 - [x] Directory Listing / Explorer
	 - [x] File Management
		 - [x] Edit Files ( Monaco Code Editor ) 
		 - [x] Delete Files / Folders
		 - [x] Create Files / Folders
		 - [x] Manage File / Folder Permissions
		 - [x] Copy Files
		 - [x] Upload Files
		 - [x] Move Files / Folders
		 - [x] Create / Extract Archives
		 - [x] Download Files
 - [x] Databases
	 - [x] List Databases
	 - [x] View Database Details
	 - [x] Rotate Database Password
	 - [x] Delete Database
	 - [x] Create Database
 - [x] Schedules
	 - [x] List Schedules
	 - [x] Create Schedules
	 - [x] Edit Schedules
	 - [x] Edit Schedule Tasks
 - [x] Users
	 - [x] List Subusers
	 - [x] Create Subusers
	 - [x] Delete Subusers
	 - [x] Edit Subusers
	 - [x] Edit Existing Subuser permissions
- [x]  Backups
	- [x] List Backups
	- [x] Create Backup
	- [x] Delete Backup
	- [x] Lock/Unlock Backup
	- [x] Download Backup
	- [x] Restore from Backup
- [x] Network
	- [x] List Allocations
	- [x] Create new Allocation
	- [x] Change Primary Allocation
- [x] Startup
	- [x] List Startup Command
	- [x] Docker Image ( dropdown + read-only if manually set ) 
	- [x] Variables
- [x] Settings
	- [x] SFTP Details
	- [x] Server Name
	- [x] Server Description
	- [x] Debug Info
	- [x] Reinstall
- [x] Activity Logs

### CODE REVIEW

File: app/Filament/App/Resources/UserResource.php [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+                                    ->schema([
+                                        Tabs\Tab::make('Console')
+                                            ->schema([
+                                                Section::make()
+                                                    ->description(trans('server/users.permissions.control_desc'))
+                                                    ->icon('tabler-terminal-2')
+                                                    ->schema([
+                                                        CheckboxList::make('control')
+                                                            ->formatStateUsing(function (User $user, Set $set) {
+                                                                $server = Filament::getTenant();
```
Comment (RMartinOscar) [inline app/Filament/App/Resources/UserResource.php:174]:
```suggestion
                                                                /** @var Server $server */
                                                                $server = Filament::getTenant();
```

**Review by ItzExotical — COMMENTED** [review]
Beautiful

File: app/Filament/App/Resources/FileResource/Pages/ListFiles.php [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+                                ->hint('Enter the new name and directory of this file or folder, relative to the current directory.')
+                                ->default(fn (File $file) => $file->name)
+                                ->required()
+                                ->live(),
+                            Placeholder::make('new_location')
+                                ->content(fn (Get $get) => resolve_path('./' . join_paths($this->path, $get('location') ?? ''))),
+                        ])
+                        ->action(function (Collection $files, $data) {
+                            $location = resolve_path(join_paths($this->path, $data('location'))); // TODO: error: Array callback must have exactly two elements
+                            $files = $files->map(fn ($file) => ['to' => $location, 'from' => $file->name])->toArray();
```
Comment (RMartinOscar) [inline app/Filament/App/Resources/FileResource/Pages/ListFiles.php:365]:
```suggestion
                            $files = $files->map(fn (File $file) => ['to' => $location, 'from' => $file->name])->toArray();
```

File: app/Filament/App/Resources/FileResource/Pages/ListFiles.php [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+
+                            Notification::make()
+                                ->title(count($files) . ' Files were moved from to ' . $location)
+                                ->success()
+                                ->send();
+                        }),
+                    BulkAction::make('archive')
+                        ->authorize(auth()->user()->can(Permission::ACTION_FILE_ARCHIVE, Filament::getTenant()))
+                        ->action(function (Collection $files) {
+                            $files = $files->map(fn ($file) => $file->name)->toArray();
```
Comment (RMartinOscar) [inline app/Filament/App/Resources/FileResource/Pages/ListFiles.php:387]:
```suggestion
                            $files = $files->map(fn (File $file) => $file->name)->toArray();
```

File: app/Filament/App/Resources/FileResource/Pages/ListFiles.php [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+
+                            Notification::make()
+                                ->title('Archive created')
+                                ->success()
+                                ->send();
+                        }),
+                    DeleteBulkAction::make()
+                        ->authorize(auth()->user()->can(Permission::ACTION_FILE_DELETE, Filament::getTenant()))
+                        ->action(function (Collection $files) {
+                            $files = $files->map(fn ($file) => $file->name)->toArray();
```
Comment (RMartinOscar) [inline app/Filament/App/Resources/FileResource/Pages/ListFiles.php:411]:
```suggestion
                            $files = $files->map(fn (File $file) => $file->name)->toArray();
```

**Review by RMartinOscar — COMMENTED** [review]
Add missing File cast

File: app/Filament/App/Pages/Settings.php [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+                                TextInput::make('database_limit')
+                                    ->label('Database Limit')
+                                    ->columnSpan(1)
+                                    ->disabled()
+                                    ->formatStateUsing(fn () => $server->databases->count() . ' of ' . $server->database_limit),
+                                TextInput::make('allocation_limit')
+                                    ->label('Allocation Limit')
+                                    ->columnSpan(1)
+                                    ->disabled()
+                                    ->formatStateUsing(fn () => $server->allocations->count() . ' of ' .  $server->allocation_limit + 1),
```
Comment (RMartinOscar) [inline app/Filament/App/Pages/Settings.php:132]:
```suggestion
                                    ->formatStateUsing(fn () => $server->allocations->count() . ' of ' .  $server->allocation_limit ?? 0 + 1),
```
https://github.com/pelican-dev/panel/blob/038eca8c4b90324517dd0fe678d442e8ed44fa24/app/Models/Server.php#L45

Comment (Boy132) [reply app/Filament/App/Pages/Settings.php]: Please don't make reviews to fix linting issues.
We are constantly working on the branch and all linting issues will be resolved eventually.

**Comment by RMartinOscar** [comment]
image
image

**Review by RMartinOscar — APPROVED** [review]
LGTM

**Pull request merged**